File: InPlan.Web/Modules/Parametros/AmbientesProyectos/AmbientesProyectosRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;
using System.ComponentModel;

namespace InPlan.Parametros;

[ConnectionKey("Default"), Module("Parametros"), TableName("AmbientesProyectos")]
[DisplayName("Ambientes Proyectos"), InstanceName("Ambiente Proyecto")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[LookupScript]
public sealed partial class AmbientesProyectosRow : Row<AmbientesProyectosRow.RowFields>, IIdRow, INameRow
{
    const string jIdProyecto = nameof(jIdProyecto);
    const string jIdAmbiente = nameof(jIdAmbiente);

    class RowTemplate
    {
        [DisplayName("Id Ambiente Proyecto"), Identity, IdProperty]
        public int? IdAmbienteProyecto { get; set; }

        [LookupEditor(typeof(ProyectosRow))]
        [DisplayName("Proyecto"), NotNull, ForeignKey("Proyectos", "IdProyecto"), LeftJoin(jIdProyecto), TextualField(nameof(IdProyectoNombreProyecto))]
        public int? IdProyecto { get; set; }

        [LookupEditor(typeof(AmbientesRow))]
        [DisplayName("Ambiente"), NotNull, ForeignKey("Ambientes", "IdAmbiente"), LeftJoin(jIdAmbiente), TextualField(nameof(IdAmbienteDescripcion))]
        public int? IdAmbiente { get; set; }

        [DisplayName("Datos de Servidor"), QuickSearch, NameProperty]
        public string ServidorDatos { get; set; }

        [DisplayName("Instacia de Servidor")]
        public string InstaciaServidor { get; set; }

        [DisplayName("Servidor Web")]
        public string ServidorWeb { get; set; }

        [DisplayName("Notas Adicionales")]
        public string NotasAdicionales { get; set; }

        [MultipleFileUploadEditor]
        [DisplayName("Documentos Relacionados")]
        public string DocumentosRelacionados { get; set; }

        [DisplayName("Observaciones")]
        public string Observaciones { get; set; }

        [DisplayName("Proyecto"), Expression($"{jIdProyecto}.[NombreProyecto]")]
        public string IdProyectoNombreProyecto { get; set; }

        [DisplayName("Ambiente"), Expression($"{jIdAmbiente}.[Descripcion]")]
        public string IdAmbienteDescripcion { get; set; }
    }
}

File: InPlan.Web/Modules/ServerTypes/Parametros/AmbientesProyectosRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface AmbientesProyectosRow {
    IdAmbienteProyecto?: number;
    IdProyecto?: number;
    IdAmbiente?: number;
    ServidorDatos?: string;
    InstaciaServidor?: string;
    ServidorWeb?: string;
    NotasAdicionales?: string;
    DocumentosRelacionados?: string;
    Observaciones?: string;
    IdProyectoNombreProyecto?: string;
    IdAmbienteDescripcion?: string;
}

export abstract class AmbientesProyectosRow {
    static readonly idProperty = 'IdAmbienteProyecto';
    static readonly nameProperty = 'ServidorDatos';
    static readonly localTextPrefix = 'Parametros.AmbientesProyectos';
    static readonly lookupKey = 'Parametros.AmbientesProyectos';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<AmbientesProyectosRow>('Parametros.AmbientesProyectos') }
    static async getLookupAsync() { return getLookupAsync<AmbientesProyectosRow>('Parametros.AmbientesProyectos') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<AmbientesProyectosRow>();
}